src/macaron/slsa_analyzer/checks/build_script_check.py

```diff
@@ -59,20 +58,22 @@ def run_check(self, ctx: AnalyzeContext, check_result: CheckResult) -> CheckResu
         CheckResultType
             The result type of the check (e.g. PASSED).
         """
-        build_tool = ctx.dynamic_data["build_spec"].get("tool")
+        build_tools = ctx.dynamic_data["build_spec"]["tools"]
 
-        # Check if a build tool is discovered for this repo.
+        if not build_tools:
+            failed_msg = "The target repository does not have any build tools."
+            check_result["justification"].append(failed_msg)
+            return CheckResultType.FAILED
+
+        # Check if any build tools are discovered for this repo.
         # TODO: look for build commands in the bash scripts. Currently
-        # we parse bash scripts that are reachable through CI only.
-        if build_tool and not isinstance(build_tool, NoneBuildTool):
-            pass_msg = f"The target repository uses build tool {build_tool.name}."
+        #       we parse bash scripts that are reachable through CI only.
+        for tool in build_tools:
+            pass_msg = f"The target repository uses build tool {tool.name}."
```

Comment:
I think we also need to check if `build_tool` is not `NoneBuildTool` here too.

Reply:
To confirm you mean as in if there is a NoneBuildTool within the list we treat that as an instant fail?

Reply:
@timyarkov Hmmm, you are right. We check for `NoneBuildTool` before because we only support 1 build tool and `ctx.dynamic_data["build_spec"]["tool"]` was originally initialized as an instance of `NoneBuildTool`. 
Your changes already set `ctx.dynamic_data["build_spec"]["tools"]` to be an empty list so this statement `represent a lack of any build tools, so wouldn't an empty list do just fine for that` holds. 
@behnazh-w Please correct me if I am wrong. I think we still need to check for NoneBuildTool here as a safeguard because printing out the result for `NoneBuildTool` wouldn't make any sense for the user :thinking: .

Reply:
Looks like `NoneBuildTool` is not instantiated anywhere in the code anymore. Is that correct @timyarkov ?

If that's true, if we remove `NoneBuildTool` class altogether, we won't need to to check for it here.

Reply:
Yeah far as I see that's the case, I'll remove it then